File: Array%20Sorting%20Algorithms/counting-sort/counting-sort.py
def counting_sort(arr):
    max_value = max(arr)
    min_value = min(arr)
    counting_arr = [0] * (max_value - min_value + 1)

    for num in arr:
        counting_arr[num - min_value] += 1

    sorted_arr = []
    for i in range(len(counting_arr)):
        for j in range(counting_arr[i]):
            sorted_arr.append(i + min_value)

    return sorted_arr

# Example usage:
arr = [64, 34, 25, 12, 22, 11, 90]
sorted_arr = counting_sort(arr)
print("Sorted array:", sorted_arr)

File: array-sorting-algorithms/counting-sort/src/src-java/counting-sort.java
public class CountingSort {
    public static int[] countingSort(int[] arr) {
        int max = Arrays.stream(arr).max().getAsInt();
        int min = Arrays.stream(arr).min().getAsInt();
        int[] countingArr = new int[max - min + 1];

        for (int num : arr) {
            countingArr[num - min]++;
        }

        int[] sortedArr = new int[arr.length];
        int sortedIndex = 0;
        for (int i = 0; i < countingArr.length; i++) {
            while (countingArr[i] > 0) {
                sortedArr[sortedIndex++] = i + min;
                countingArr[i]--;
            }
        }

        return sortedArr;
    }

    // Example usage:
    public static void main(String[] args) {
        int[] arr = {64, 34, 25, 12, 22, 11, 90};
        int[] sortedArr = countingSort(arr);
        System.out.println("Sorted array: " + Arrays.toString(sortedArr));
    }
}
